packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -641,9 +683,12 @@ class _TooltipOverlay extends StatelessWidget {
               child: Center(
                 widthFactor: 1.0,
                 heightFactor: 1.0,
-                child: Text(
-                  message,
-                  style: textStyle,
+                child: RichText(
+                  text: richMessage,
+                  textScaleFactor: (context.widget is MediaQuery ||
```

Comment:
this was added so that the tooltip text will scale with the MediaQuery textScaleFactor as expected. Without this change, the corresponding test that verifies this functionality was failing.

Reply:
We should use Text.rich(..) which will grab the textScaleFactor from the context automatically.